.env.example

```
@@ -41,6 +51,22 @@ LLM_MAX_TOKENS=2048
 AGENT_MAX_TOKENS=4096
 # End-to-end timeout for /agent/query in seconds
 AGENT_TIMEOUT=120
+# Thinking budget for ALL agentic-mode LLM calls (query planner, tool routing,
+# query expansion, answer generation, reflexion judge).
+#   0  = thinking OFF   (cheapest — default; lowest API bill)
+#   -1 = DYNAMIC        (model decides; smarter routing/answers, billed thinking tokens)
+#   N  = cap to N tokens of thinking per call (e.g. 512)
+# Raise only if agent quality is the bottleneck, not cost.
+AGENT_THINKING_BUDGET=-1
```

dotenv-linter (4.0.0)

[warning] [UnorderedKey] The AGENT_THINKING_BUDGET key should go before the AGENT_TIMEOUT key

(UnorderedKey)